src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,116 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int CAPACITY = 16;
+    private static final int MAX_CAPACITY = Integer.MAX_VALUE / 2;
+    private static final float LOAD_FACTOR = 0.75f;
+    private int size = 0;
+    private Node<K, V>[] hashtable;
+
+    public MyHashMap() {
+        hashtable = new Node[CAPACITY];
+    }
+
+    private class Node<K, V> {
+        private K key;
+        private V value;
+        private Node<K, V> next;
+
+        private Node(K key, V value) {
+            this.key = key;
+            this.value = value;
+            next = null;
+        }
+
+        private K getKey() {
+            return key;
+        }
+
+        private V getValue() {
+            return value;
+        }
+
+        private Node<K, V> getNext() {
+            return next;
+        }
+
+        private void setValue(V value) {
+            this.value = value;
+        }
+    }
+
+    private int getIndex(K key) {
+        return key == null ? 0 : Math.abs(key.hashCode()) % hashtable.length;
+    }
+
+    private void resizeMap() {
+        int capacity = hashtable.length;
+        if (capacity == MAX_CAPACITY) {
+            return;
+        }
+        if (capacity << 1 >= MAX_CAPACITY) {
+            capacity = MAX_CAPACITY;
+        } else {
+            capacity = capacity << 1;
+        }
+        Node<K, V>[] oldHashTable = hashtable;
+        hashtable = new Node[capacity];
+        size = 0;
+        for (Node<K, V> node : oldHashTable) {
+            if (node != null) {
+                while (node != null) {
+                    put(node.key, node.value);
+                    node = node.next;
+                }
+            }
+        }
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= hashtable.length * LOAD_FACTOR) {
```

Comment:
let's create variable threshold at the beginning 
```suggestion
        if (size >= threshold) {
```